sqlmesh/core/snapshot/definition.py

```diff
@@ -888,35 +886,34 @@ def missing_intervals(
         )
 
         interval_unit = self.node.interval_unit
+        upper_bound_ts = to_timestamp(execution_time or now())
 
-        execution_time = execution_time or now()
-        if end_bounded:
-            execution_time = min(to_timestamp(execution_time), end_ts)
-
-        if not allow_partials:
-            upper_bound_ts = to_timestamp(
-                self.node.cron_floor(execution_time) if not ignore_cron else execution_time
-            )
-            end_ts = min(end_ts, to_timestamp(interval_unit.cron_floor(upper_bound_ts)))
-        else:
-            upper_bound_ts = to_timestamp(execution_time)
+        if allow_partials or ignore_cron:
             end_ts = min(end_ts, upper_bound_ts)
+        else:
+            end_ts = min(
```

Comment:
These kinds of deeply nested expressions can really benefit from a comment explaining the intention

Reply:
i'm not really sure what to say, it's kind of self explanatory